<center>
    <a href="https://www.aims-senegal.org/" ><img src="aimssn_logo.png" style="float:center; max-width: 600px; display: inline" alt="IMSP"/> </a>
    </center>
    
#    <center> <b> <span style="color:orange;" >  `Python for Scientific Computing and Software Engineering ` </span> </b></center>


## <center> <b> <span style="color:orange;"> An Initiation to Programming using Python (Init2Py) </span> </b></center>

### <center> <b> <span style="color:blue;"> Practicals: Variables, Types & Operations </span> </b></center>

**Exercise 00:**  

> 1. Declare a variable `firstname` and assign it a value.
> 2. Declare a `lastname` variable and assign a value to it.
> 3. Declare a `full_name` variable and assign it a value.
> 4. Declare a `country` variable and assign it a value.
> 5. Declare a variable `city` and assign it a value.
> 6. Declare a `age` variable and assign it a value.
> 7. Declare a variable `year` and assign it a value.
> 8. Declare a variable `is_marrie` and assign it a value.
> 9. Declare a variable `is_true` and assign it a value.

> For each of the above variable, display the value it contains using the `print()` function.




In [1]:
firstname="DONGMO TEDO"
lastname="Lynsia Saychele"
full_name= "DONGMO TEDO Lynsia Saychele"
country="CAMEROON"
city="Dschang"
age= 23
year= 2021
is_marrie=False
is_true= True
print(firstname,lastname,full_name,country,city,age,year,is_marrie,is_true)







DONGMO TEDO Lynsia Saychele DONGMO TEDO Lynsia Saychele CAMEROON Dschang 23 2021 False True


**Exercise 01:** 
> Check the data type of all your variables using the built-in function `type()`.**


In [2]:
type(firstname)
type(lastname)
type(full_name)
type(country)
type(city)
type(year)
type(is_marrie)
type(is_true)


bool

**Exercise 02:** 
> Calculate the value of $y$ ($y = x^2 + 6x + 9$). Try using different values of $x$ and determine at what value of $x$ will $y$ be equal to 0.**

In [3]:
for x in [1,2,-3]:
    y=x**2+6*x+9
    print('x:',x,'y:',y)

x: 1 y: 16
x: 2 y: 25
x: -3 y: 0


In [4]:
list(range(-3,5))
y=x**2+6*x+9
print('x:',x,'y:',y)

x: -3 y: 0


In [5]:
x=-3
y=x**2+6*x+9
print('x:',x,'y:',y)
    

x: -3 y: 0


**Exercise 03:** 
>    
     1. Declare your age as an integer variable
     2. Declare your height as a float variable
     3. Declare a variable that stores the base and height of a triangle. Calculate the area of this triangle (area = 0.5 x b x h).

In [6]:
myage =23
height=2.2
stockes= (5, 1.5)
area = 0.5*stockes[0]*stockes[1]
print(area)

3.75


## *Tip: receiving input from the user *
**To receive an input from the user, we use `input()`.**.

**Example**
Receive two numbers as input from the user and calculate their power:
```python
base = input("Enter a base: ")
exponent = input("Enter an exponent: ")
result = float(base) ** float(exponent)
print(f "{base} to the power of {exponent} = {result}")
```

**Exercise 04:** 
> **Write a small code that asks the user to enter the side $a$, the side $b$ and the side $c$ of a triangle. Calculate the perimeter of the triangle (perimeter = a + b + c).**

In [ ]:
a= float(input("entrer la valeur  de a"))
b=float(input("entrer la valeur de b"))
c=float(input("entrer la valeur de c"))
p=a*b*c
print(p)

**Exercise 05:** 
> **Obtain the length and width of a rectangle using the prompt. Calculate its `area`and its `perimeter`.**

In [2]:
L=float(input("entrer la longeur :"))
l=float(input("entrer la largeur :")) 
p=(l+L)*2
a=l*L
print('Le perimetre est',p)
print('aire est',a)
    

entrer la longeur :2
entrer la largeur :3
Le perimetre est 10.0
aire est 6.0


**Exercice 06:** 
> **Obtenez le rayon d'un cercle en utilisant l'invite. Calculez l'aire (aire = pi x r x r) et la circonférence (c = 2 x pi x r) où pi = 3,14.**

In [7]:
r=float(input("entrer le rayon du cercle"))
pi=3.14
aire=pi*r*r
c=2*pi*r
print('aire est',a)
print('la circonference est',c)

entrer le rayon du cercle2
aire est 6.0
la circonference est 12.56


**Exercice 07:**
> **Calculez la pente, l'ordonnée à l'origine en x et l'ordonnée à l'origine en $y$ de $y = 2x -2$.**

In [9]:
x1=float(input("entrer la valeur de x1"))
x2=float(input("entrer la valeur de x2"))
y2=2*x1-2
y1=2*x2-2
a=(y2-y1)/(x2-x1)
x=0
y0=2*x-2
y=0
x0=(y-2)/2
print('la pente est',a)
print('ordonne origine en x est',x0)
print('ordonne origine en y est',y0)

entrer la valeur de x15
entrer la valeur de x22
la pente est -2.0
ordonne origine en x est -1.0
ordonne origine en y est -2


**Exercice 08:** 
> **Rédigez un script qui demande à l'utilisateur d'entrer les heures et le taux horaire. Calculez la rémunération de la personne ?**

In [13]:
heure=float(input("entrer la valeur de l'heure"))
t=float(input("entre la valeur du taux horaire"))
r=t*heure
print('la remuneration est',r)


entrer la valeur de l'heure5
entre la valeur du taux horaire25
la remuneration est 125.0


**Exercice 09:** 
> **Écrivez un script qui demande à l'utilisateur d'entrer le nombre d'années. Calculez le nombre de secondes qu'une personne peut vivre. Supposons qu'une personne puisse vivre cent ans.**

In [21]:
a=float(input("entrer le nombre d'annee"))
b=365
print('le nombre de jours correspondant a 1an est',b)
c=24
print('le nombre heure correspondant a 1jour est',c)
d=60
print('le nombre minute correspondant a 1heure est',d)
e=60
print('le nombre seconde correspondant a 1minute est',e)
f=(100-a)*b*c*d*e
print('le de seconde une personne puisse vivre est',f)



entrer le nombre d'annee2
le nombre de jours correspondant a 1an est 365
le nombre heure correspondant a 1jour est 24
le nombre minute correspondant a 1heure est 60
le nombre seconde correspondant a 1minute est 60
le de seconde une personne puisse vivre est 3090528000.0


**Exercice 10:** 
> **Écrivez un script Python qui affiche le tableau suivant**

    1 1 1 1 1
    
    2 1 2 4 8
    
    3 1 3 9 27
    
    4 1 4 16 64
    
    5 1 5 25 125

In [50]:
k=1
print(k,k,k,k,k)
print(k+1,k,(k+1)**2,(k+1)**3)
print(k+2,k,k+2,(k+2)**2,(k+2)**3)
print(k+3,k,k+3,(k+3)**2,(k+3)**3)
print(k+4,k,k+4,(k+4)**2,(k+4)**3)

1 1 1 1 1
2 1 4 8
3 1 3 9 27
4 1 4 16 64
5 1 5 25 64


**Exercise 11:** 
>**Assign the variables `time` and `distance` with the values $6.892$ and $19.7$. Calculate and display the value of the speed. Improve the display by imposing a digit after the decimal point.**

In [24]:
t=6.892
d=19.7
v= d/t
 
print('la valeur de la vitesse est:',round (v,1))

la valeur de la vitesse est: 2.9


**Exercice 12:** 
> **Écrire un programme qui, à partir de la saisie d’un rayon et d’une hauteur, calcule levolume d’un cône droit.**

In [27]:
r=float(input("entrer la valeur du rayon"))
h=float(input("entrer la valeur de la hauteur"))
pi=3.14
B=pi*r*r
V=1/3*(B*h)  
print('le volume du cone est:',V)        
        
        

entrer la valeur du rayon2
entrer la valeur de la hauteur5
le volume du cone est: 20.933333333333334


**Exercise 13:** 
> **The user gives a positive integer and the program announces how many times in a row this integer is divisible by 2.**

In [49]:
a=int(input("entrer un entier positif"))
x=0
while(a%2==0):
    a/=2
    x=x+1
print('le nombre de fois que a divise 2 est: ',x)


entrer un entier positif8
le nombre de fois que a divise 2 est:  3


Exercice 14:

        **The user gives an integer between 2 and 12, the program gives the number of ways to do this by rolling two dice. 

In [53]:
a=int(input("entrer un nombre entier entre 2 et 12: "))
k=0
for p in range (1,7):
    for q in range (1,7):
        if p+q==a:
            k=k+1
print('le nombre de facon de faire est: ',k)
        

entrer un nombre entier entre 2 et 12: 5
le nombre de facon de faire est:  4


**Exercice 15:** 
> **Utilisez la fonction d'entrée intégrée `input()` pour obtenir le `_prénom_`, le `_nom_`, le `_pays_d_origine` et l'`_age_` d'un utilisateur et stockez la valeur dans les noms des variables correspondantes.**

In [ ]:
prenom= input("entrer le prenom: ")
nom=input("entrer le nom: ")
pays_origine=input("entrer pays origine: ")
age=input("entrer age: ")


**Exercice 16:** 

**Déclarer $5$ comme `num_one` et $4$ comme `num_two`**

   1. Additionner `num_one` et `num_two` et affecter la valeur à une variable `totale`.
   2. Soustraire `num_two` de `num_one` et affecter la valeur à une variable `diff`.
   3. Multipliez le `num_one` et le `num_two` et attribuez la valeur à une variable `produit`.
   4. Divisez le `num_one` par le `num_two` et attribuez la valeur à une variable `division`.
   5. Calculez `num_one` à la puissance de `num_two` et attribuez la valeur à une variable `exp`.

In [36]:
num_one=5
num_two=4
total=num_one+num_two
print(total)
diff=num_two-num_one
print(diff)
produit=num_two*num_one
print(produit)
division=num_one/num_two
print(division)
exp=num_one**num_two
print(exp)

9
-1
20
1.25


<center> <b> <span style="color:orange;"> 🎉 FÉLICITATIONS! / CONGRATULATIONS! 🎉 </span> </b></center>